Automattic/wp-calypso
# Checkout: Move Giropay to wpcom-checkout package

#### Changes proposed in this Pull Request

This PR moves the Giropay payment method from the `@automattic/composite-checkout` package into the `@automattic/wpcom-checkout` package. This is similar to how Bancontact was moved in https://github.com/Automattic/wp-calypso/pull/51690

<img width="582" alt="Screen Shot 2021-06-02 at 1 45 05 PM" src="https://user-images.githubusercontent.com/2036909/120527817-c9a6b580-c3a8-11eb-821c-bfb33ec6fc8a.png">


The `composite-checkout` package is intended to be a fairly generic checkout-building toolkit. As part of this we originally envisioned it containing various payment methods, but that quickly became a problem as it became clear that the payment methods we use on WordPress.com are pretty specific to our platform. To reduce leaky abstractions, we moved many of our payment methods over to calypso, but some have remained because the cost of moving them seemed too high for the benefit.

As we prepare to publish version 1.0.0 of the `@automattic/composite-checkout` package on npm, I think it's a good time to clean up extraneous parts of the package. The payment methods are the only part of the package that are not fully TypeScript and are also still pretty specific to our WordPress.com infrastructure. Eventually it would be nice to convert them all to TypeScript and put them in a package like `@automattic/wpcom-checkout` (which is intended to be WordPress.com specific). This PR converts one such payment method and moves it there.

#### Testing instructions

- Apply D45434-code to your sandbox to force Giropay to be available.
- Set your user currency to EUR.
- Make a new purchase in checkout and select Giropay as the payment method, entering any string in the "name" field.
- Verify that you are redirected to the Stripe test page, that it reads "Giropay", and that the total is correct. (No need to complete the payment.)
